vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/PollItem.kt

```diff
@@ -73,13 +72,21 @@ abstract class PollItem : AbsMessageItem<PollItem.Holder>() {
         optionViewStates.forEachIndexed { index, optionViewState ->
             views.getOrNull(index)?.let {
                 it.render(optionViewState)
-                it.setOnClickListener {
-                    callback?.onTimelineItemAction(RoomDetailAction.VoteToPoll(relatedEventId, optionViewState.optionId))
-                }
+                it.setOnClickListener { onPollItemClick(optionViewState) }
             }
         }
     }
 
+    private fun onPollItemClick(optionViewState: PollOptionViewState) {
+        val relatedEventId = eventId
+
+        if (isPollActive(optionViewState) && relatedEventId != null) {
+            callback?.onTimelineItemAction(RoomDetailAction.VoteToPoll(relatedEventId, optionViewState.optionId))
+        }
+    }
+
+    private fun isPollActive(optionViewState: PollOptionViewState) = optionViewState !is PollOptionViewState.PollEnded
```

Comment:
There is too much logic here. Better to move that to the controller.
Maybe replace `var pollSent: Boolean = false` which is unused by something like `var canVote: Boolean = false`